### ISSUE TITLE
[Event Hubs Client] Enable Tests to Manage Service Needs

### ISSUE THREAD

**Issue by jsquire** [Azure/azure-sdk-for-net#5987]
# Summary

In its current implementation, the live tests in the Event Hubs suite assume a shared environment that exists at the time of execution.  This limits the ability of tests to run in parallel and increases the risk that failing or misbehaved tests will influence other tests.

In order to ensure isolation and deterministic results of tests, each test should assume the minimum prepared Azure environment consisting only of a namespace, storage account, and service principal and should take responsibility for creating/removing any additional constructs needed for their execution in a dedicated scope.

# Scope of Work

- Enable creating and deletion of entities on a given Event Hubs namespace using the management library or other approach.

- Each live test should create any entities within the namespace that it needs, use them, and tear them down when it has completed.  Tear down should be performed regardless of test outcome or the presence of an exception.

- The `ReadMe` and associated ARM templates or other assets should be updated to capture changes needed to preparing the environment, removing creation of the shared event hub and including any new service principals or other assets needed.

# Out of Scope

- Refactoring of the general approach for testing, the test cases, code style, and other artifacts of the codebase; changes should be kept as tightly scoped and as minimal as possible.

- Adding of additional test cases or new testing scenarios.

# Success Criteria

- Tests should be fully isolated in their use of Event Hubs entities; tests should not impact each other due to shared entities.

- Tests should have responsibility for creating and tearing down any entities used, and should be tolerant of test failures and exceptions when doing so; exception cases should not result in leaking test entities.

- Consumers that wish to clone the repository and execute the tests should have clear instructions on how to do so, and automation should be provided in as few steps as possible.  Any automation should be tolerant of being run across the Windows, Linux, and macOS platforms.

- Consumers should be able to prepare an Azure environment to run live tests in 5 steps or less.
